spec/factories/furniture/marketplace.rb

```diff
@@ -131,7 +131,7 @@
   end
 
   factory :marketplace_order, class: "Marketplace::Order" do
-    marketplace
+    marketplace { association(:marketplace, :ready_for_shopping) }
```

Comment:
Do all `Orders` need a `ready_for_shopping` markeplace? FactoryBot encourages making the baseline factory as minimal as possible, and use a trait if you need something more.

Reply:
Considering that principle then I suppose no. But following that we'd need a variation like `marketplace_order_ready_for_shopping`whatever for use with the mailer in https://github.com/zinc-collective/convene/blob/main/spec/mailers/previews/marketplace/order/received_mailer_preview.rb#L5. I'm thinking it may be a worthwhile exercise to write a new factory stack used by the mailer to see if we can fully simulate the correct data model and compare to the base factories currently there. Maybe it's a special that is only used for the mailer preview, then we don't have to worry so much about recreating the entire data model with the base factories?

Reply:
I don't think it makes sense to add a `marketplace_order_for_mailer_preview` or rewrite the factory stack. 

We have two pretty easy options for how to do this without promoting `Order`'s default factory to include a Marketplace that is :ready_for_shopping.


Option A) Instantiate the Marketplace where we need it
```rb
class Marketplace
  class Order
    class ReceivedMailerPreview < ActionMailer::Preview
      def notification
        marketplace = create(:marketplace, :ready_for_shopping)
        Order::ReceivedMailer.notification(FactoryBot.build(:marketplace_order, :full, marketplace:))
      end
    end
  end
end
```

Option B) Update the [`:full` trait to include a `Marketplace` generated with the `:ready_for_shopping` trait.](https://github.com/zinc-collective/convene/blob/main/spec/factories/furniture/marketplace.rb#L168) (which apparently means finding out why `the `marketplace :full` trait isn't :ready_for_shopping...)